mist-common/src/main/java/edu/snu/mist/common/operators/FixedSizeWindowOperator.java

```diff
@@ -143,17 +150,39 @@ protected void putWatermark(final MistWatermarkEvent input) {
     }
     if (input.isCheckpoint()) {
       latestCheckpointTimestamp = input.getTimestamp();
+      final Map<String, Object> stateMap = new HashMap<>();
+      stateMap.put("windowCreationPoint", windowCreationPoint);
+      stateMap.put("windowQueue", windowQueue);
+      checkpointMap.put(input.getTimestamp(), stateMap);
     }
   }
 
   @Override
-  public Map<String, Object> getOperatorState() {
+  public Map<String, Object> getCurrentOperatorState() {
     final Map<String, Object> stateMap = new HashMap<>();
     stateMap.put("windowCreationPoint", windowCreationPoint);
     stateMap.put("windowQueue", windowQueue);
     return stateMap;
   }
 
+  @Override
+  public Map<String, Object> getOperatorState(final long timestamp) {
```

Comment:
The codes `getOperatorState` and `removeStates` are duplicate. I think we need to refactor this code.